modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -0,0 +1,502 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BaseTransactionBuilder, TransactionType } from '../baseCoin';
+import { InvalidTransactionError, BuildTransactionError } from '../baseCoin/errors';
+import { BaseAddress, BaseKey } from '../baseCoin/iface';
+import { isValidEd25519Seed } from '../../utils/crypto';
+import { testnetMetadataRpc, westendMetadataRpc, mainnetMetadataRpc } from './metadataRpc';
+import BigNumber from 'bignumber.js';
+import { decodeAddress } from '@polkadot/keyring';
+import { getRegistry, decode } from '@substrate/txwrapper-polkadot';
+import { UnsignedTransaction } from '@substrate/txwrapper-core';
+import { TypeRegistry, DecodedSignedTx, DecodedSigningPayload } from '@substrate/txwrapper-core/lib/types';
+import { BaseTransactionSchema, SignedTransactionSchema, SigningPayloadTransactionSchema } from './txnSchema';
+import { Transaction } from './transaction';
+import { KeyPair } from './keyPair';
+import { CreateBaseTxInfo, sequenceId, specNameType, TxMethod, Validity } from './iface';
+import Utils from './utils';
+import { AddressValidationError } from './errors';
+
+export abstract class TransactionBuilder extends BaseTransactionBuilder {
```

Comment:
May be we should include functionality to get the fee rate for transactions.
https://wiki.polkadot.network/docs/learn-transaction-fees#fee-adjustment

Reply:
fee rate would need connection to the chain, [Fee Adjustment](Fees increase or decrease for the next block based on the fullness of the current block relative to the target). How should we get the fees offline?

Reply:
Agree @gibrangul95  I'll do a double check. May be this funcionality should be on the platform. Thank you.